src/main/java/com/commercetools/sync/commons/utils/CommonTypeUpdateActionUtils.java

```diff
@@ -0,0 +1,31 @@
+package com.commercetools.sync.commons.utils;
+
+
+import com.commercetools.sync.commons.helpers.SyncResult;
+import io.sphere.sdk.commands.UpdateAction;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.Objects;
+
+public class CommonTypeUpdateActionUtils {
+
+    /**
+     * Compares two {@link Object} and returns a {@link SyncResult} containing the supplied {@link UpdateAction} as a
+     * result. If no update action is needed, for example in case where both the {@link Object}
+     * have the same values, a sync result with containing an empty list of update actions is returned.
+     *
+     * @param oldObject    the object which should be updated.
+     * @param newObject    the object with the new information.
+     * @param updateAction the update action to return in the optional.
+     * @return A sync result with a list containing the update action or a sync result containing an empty list
+     * of update actions if the object values are identical.
+     */
+    @Nonnull
+    public static <T> SyncResult<T> buildUpdateAction(@Nullable final Object oldObject,
```

Comment:
Seems to be confusing to return SyncResult which implies "after sync" result and not building update actions. Also i am not sure about of benefit of this wrapper as it really belongs to the class which does the sync process only and not all the possible utils across module.